### DISCUSSION TITLE
Is there currently a way to create multiple instances?

### DISCUSSION THREAD

**Discussion by nedredmond** [zorkow/speech-rule-engine#747] [General]
I am currently using the Speech Rule Engine in a few different use cases-- one being automated aria labels, for which `clearspeak` works great.

However, I have another use case where I believe I should be using `mathspeak` in `verbose` mode-- that being use in a MathQuill math expression editor. I'm a bit concerned that since there is only one instance of SRE for the whole application, there will be bleedover and confusion if I call setup separately for these different purposes.

Is there any known pattern for getting different modes on the fly or per `toSpeech` expression? Thanks!

**Comment by kgryte** [comment]
I also agree that it would be beneficial to move away from the singleton pattern and global state to being able to create multiple instances. For example, when wanting to generate static output for multiple modalities.
